libfreerdp/core/peer.c

```diff
@@ -1225,146 +1225,78 @@ static BOOL freerdp_peer_send_channel_data(freerdp_peer* client, UINT16 channelI
 	return rdp_send_channel_data(client->context->rdp, channelId, data, size);
 }
 
+static BOOL redir_write_string(UINT32 flag, wStream* s, const char* str)
+{
+	const size_t length = (strlen(str) + 1);
+	if (!Stream_EnsureRemainingCapacity(s, 4ull + length * sizeof(WCHAR)))
+		return FALSE;
+
+	Stream_Write_UINT32(s, length);
```

Comment:
This field is the length in bytes so if we are writing in UTF16 that should be `length * sizeof(WCHAR)` isn't it?